src/index.ts

```diff
@@ -2249,6 +2256,191 @@ async function claimAllUnclaimedAsSenderPerChain({
 	return txHashes
 }
 
+// Returns args to be passed to makeDepositWithAuthorization function
+// and a EIP-712 message to be signed
+async function makeGaslessDepositPayload({
+	address,
+	contractVersion,
+	linkDetails,
+	password,
+}: interfaces.IPrepareGaslessDepositParams): Promise<interfaces.IMakeGaslessDepositPayloadResponse> {
+	if (!PeanutsWithEIP3009.includes(contractVersion)) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: this Peanut version does not support gasless deposits'
+		)
+	}
+
+	if (linkDetails.tokenType !== interfaces.EPeanutLinkType.erc20) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: only erc20 tokens are currently supported for gasless deposits'
+		)
+	}
+
+	if (!linkDetails.tokenAddress) {
+		throw new interfaces.SDKStatus(
+			interfaces.EPrepareCreateTxsStatusCodes.ERROR_VALIDATING_LINK_DETAILS,
+			'Error validating link details: token address must be provided'
+		)
+	}
+
+	if (!linkDetails.tokenDecimals) {
+		linkDetails.tokenDecimals = 18
+	}
+
```

Comment:
??

I don't think this is a safe assumption. USDC has 6 decimals. Most erc20s either have 18 or 6, but it could be anything.

Either we force user to supply tokenDecimals, or we can get it through RPC call from onchain too